parity/lib.rs

```diff
@@ -141,6 +141,35 @@ fn print_hash_of(maybe_file: Option<String>) -> Result<String, String> {
 	}
 }
 
+#[cfg(feature = "deadlock_detection")]
+fn run_deadlock_detection_thread() {
+	use std::thread;
+	use std::time::Duration;
+	use parking_lot::deadlock;
+
+	info!("Starting deadlock detection thread.");
+	// Create a background thread which checks for deadlocks every 10s
+	thread::spawn(move || {
+		loop {
+			thread::sleep(Duration::from_secs(10));
+			let deadlocks = deadlock::check_deadlock();
+			if deadlocks.is_empty() {
+				continue;
+			}
+
```

Comment:
Grumble on this if statement because deadlock iterator will handle this case but I guess it is ok if you want to print the number of deadlocks!

No big deal for me I just want the code to be as simple as possible!